source/production/F0.Minesweeper.Logic/LocationShuffler/ILocationShuffler.cs

```diff
@@ -0,0 +1,11 @@
+using System.Collections.Generic;
+using F0.Minesweeper.Logic.Abstractions;
+
+namespace F0.Minesweeper.Logic.LocationShuffler
+{
+	internal interface ILocationShuffler
+	{
+		internal static ILocationShuffler Instance { get; } = default!;
```

Comment:
I guess the idea was to enforce all `ILocationShuffler` implementations to provide a singleton instance. This way we only have one instance and don't have to create a new shuffler each time a new game is started. The idea makes sense to me since the shuffler implementations should have no state anyway.

It could be executed a bit better via an `abstract` base class. Basically something like this:
```
internal abstract class LocationShufflerBase<T> where T : ILocationShuffer, new()
{
    internal static ILocationShuffer Default { get; } = new T();
}
```

What do you think about that?

Reply:
As discussed:
* new enum for LocationShuffler
* enum in MinefieldOptions
* enum -> objects in MinefieldFactory
* Minefield c'tor with objects for Minelayer and LocationShuffler
* Minefield remove `MinefieldOptions minefieldOptions` c'tor
* Factory has Dictionary for lazy Shuffler Instances